sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/AppManagedUserAttributeController.java

```diff
@@ -0,0 +1,323 @@
+        @ApiResponse(
+            responseCode = "404",
+            description = "User not found",
+            content = {@Content(mediaType = "application/json")}),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Invalid combinaison of id between body and path",
+            content = {@Content(mediaType = "application/json")})
+      })
+  public ResponseEntity<?> addUserAttributesManagedByApp(
+      @Parameter(
+              description = "Name of the realm where the operation will be made",
+              required = true)
+          @PathVariable("realm")
+          String realm,
+      @Parameter(
+              description = "Name of the userstorage where the operation will be made",
+              required = true)
+          @PathVariable("storage")
+          String storage,
+      @Parameter(description = "User's id to update", required = true) @PathVariable("id")
+          String id,
+      @Parameter(description = "key of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-name")
+          String attributeKey,
+      @Parameter(description = "value of attribute to add", required = true)
+          @PathVariable("app-managed-attribute-value")
+          String attributeValue) {
+
+    Realm _realm = realmProvider.load(realm);
+    if (attributeKey
+        .toUpperCase()
+        .equals(
+            _realm.getProperties().get(GlobalKeysConfig.APP_MANAGED_ATTRIBUTE_KEY).toUpperCase())) {
+
+      if (permissionService.isWriter(realm, storage)) {
+        userService.addAppManagedAttribute(realm, storage, id, attributeValue);
+        return ResponseEntity.status(HttpStatus.OK).body(userService.findById(realm, storage, id));
+      } else {
+
+        List<String> regexpAttributesAllowed =
+            permissionService.getAllowedAttributePattern(
+                realm,
+                storage,
+                _realm.getProperties().get(GlobalKeysConfig.APP_MANAGED_ATTRIBUTE_PATTERN));
+        // Check if attribute match with allowed pattern
+        for (String regexpAttributeAllowed : regexpAttributesAllowed) {
+          if (attributeValue.toUpperCase().matches(regexpAttributeAllowed.toUpperCase())) {
+            userService.addAppManagedAttribute(realm, storage, id, attributeValue);
+            return ResponseEntity.status(HttpStatus.OK)
+                .body(userService.findById(realm, storage, id));
+          }
+        }
+        // If no match found then app cannot managed attribute or attribute doesn't math
+        // with allowed pattern
+        throw new AppCannotManagedAttributeException(
+            "Cannot add attribute to user: attribute doesn't match with pattern");
+      }
+    }
+    throw new AppCannotManagedAttributeException(
+        "Cannot add attribute to user: app cannot managed " + attributeKey);
+  }
+
+  @DeleteMapping(
+      value = {"/realms/{realm}/storages/{storage}/users/{id}/app-managed-attribute"},
```

Comment:
value = {"/realms/{realm}/users/{id}/{app-managed-attribute-name}/{app-managed-attribute-value}"},
 ?

Reply:
if this is ok I don't understand why, can you explain ?

Reply:
it's an error i forgot to correct it